pallets/moonbeam-lazy-migrations/src/lib.rs

```diff
@@ -183,5 +202,46 @@ pub mod pallet {
 			}
 			Ok(Pays::No.into())
 		}
+
+		// TODO(alexandru): This extrinsic should be removed once Gov V1 is removed.
+		// Note: We don't need to unreserve any funds, as they are assumed to be already
+		// unreserved prior to this operation and the proposal submission disabled.
+		#[pallet::call_index(2)]
+		#[pallet::weight(
+			Weight::from_parts(0,
+				INTERMEDIATES_NODES_SIZE + MAX_UNLOCK_PROOF_PER_ACCOUNT * <u64>::from(*limit))
+				.saturating_add(<T as frame_system::Config>::DbWeight::get()
+				.reads_writes((*limit + 1).into(), (*limit + 1).into()).saturating_mul(2))
+		)]
+		pub fn unlock_democracy_funds(
+			origin: OriginFor<T>,
+			limit: u32,
+		) -> DispatchResultWithPostInfo {
+			ensure_signed(origin)?;
+			ensure!(limit != 0, Error::<T>::LimitCannotBeZero);
+			ensure!(limit <= 50, Error::<T>::UnlockLimitTooHigh);
+
+			ensure!(
+				!DemocracyLocksMigrationCompleted::<T>::get(),
+				Error::<T>::AllDemocracyFundsUnlocked
+			);
+
+			// Unlock staked funds and remove the voting entry. This way we can keep track of what
+			// is left without extra cost.
+			let unlocked_accounts = VotingOf::<T>::iter()
+				.drain()
+				.take(limit as usize)
+				.map(|(account, _)| {
+					<T as pallet_democracy::Config>::Currency::remove_lock(DEMOCRACY_ID, &account)
+				})
+				.count() as u32;
+
+			if unlocked_accounts < limit {
```

Comment:
Shouldn't we check for lower than or equal (`<=`) here? Otherwise if we are in the specific case where the `limit` and the amount of `unlocked_accounts` are the same, the boolean would not be set and the migration would seem to not have completed when actually it did.

Reply:
It's more tricky, if unlocked_accounts == limit, we don't know if the migration was completed or not, we need to do at least one more iteration to figure out that so the current condition is better

Reply:
Got it, looks good then.